designer/server/src/test/scala/pl/touk/nussknacker/ui/process/ProcessStateDefinitionServiceSpec.scala

```diff
@@ -72,51 +61,90 @@ class ProcessStateDefinitionServiceSpec extends AnyFunSuite with Matchers {
-                                                      ) extends OverridingProcessStateDefinitionManager(
-    customStateDefinitions = Map(
-      "COMMON" -> StateDefinitionDetails(
-        displayableName = displayableName, icon = icon, tooltip = tooltip, description = description
-      ),
-      "CUSTOM_STREAMING" -> StateDefinitionDetails(
-        displayableName = "Streaming", icon = None, tooltip = None, description = Some("This definition is specific for stremaing")
-      ),
-    )
-  )
+object ProcessStateDefinitionServiceSpec {
 
-  private class FraudProcessStateDefinitionManager(displayableName: String = "Common",
-                                                   icon: Option[URI] = None,
-                                                   tooltip: Option[String] = None,
-                                                   description: Option[String] = Some("This definition is common for all processing types")
-                                                  ) extends OverridingProcessStateDefinitionManager(
-    customStateDefinitions = Map(
-      "COMMON" -> StateDefinitionDetails(
-        displayableName = displayableName, icon = icon, tooltip = tooltip, description = description
-      ),
-      "CUSTOM_FRAUD" -> StateDefinitionDetails(
-        displayableName = "Fraud", icon = None, tooltip = None, description = Some("This definition is specific for Fraud")
-      ),
-    )
+  private val categoryConfig = ConfigFactory.parseString(
+    s"""
+       |{
+       |  categoriesConfig: {
+       |    "$Category1": "$Streaming",
+       |    "$Category2": "$Streaming",
+       |    "$TestCat": "$Fraud",
+       |    "$TestCat2": "$Fraud"
+       |  }
+       |}
+       |""".stripMargin)
+
+  private val categoryService = new ConfigProcessCategoryService(categoryConfig)
+
+  private val emptyStateDefinitionManager = new ProcessStateDefinitionManager {
+    override def stateDefinitions: Map[StatusName, StateDefinitionDetails] = Map.empty
+    override def statusActions(stateStatus: StateStatus): List[ProcessActionType] = Nil
+    override def mapActionToStatus(stateAction: Option[ProcessActionType]): StateStatus = FailedStateStatus("dummy")
+  }
+
+  def createStateDefinitionManager(definitions: Map[String, String]) = new OverridingProcessStateDefinitionManager(
+    customStateDefinitions = definitions.map { case (name, displayableName) =>
+      name -> StateDefinitionDetails(
+        displayableName = displayableName, icon = None, tooltip = None, description = Some(s"Description for ${displayableName}")
+      )
+    },
+    delegate = emptyStateDefinitionManager
   )
 
+  def testStateDefinitions(user: LoggedUser, streamingProcessStateDefinitionManager: OverridingProcessStateDefinitionManager, fraudProcessStateDefinitionManager: OverridingProcessStateDefinitionManager): List[UIStateDefinition] = {
```

Comment:
these methods can be private.

Reply:
I've moved methods from companion object and made them private.